cmd/bcpd/app/msg.go

```diff
@@ -0,0 +1,56 @@
+package app
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/x/batch"
+)
+
+var _ batch.Msg = (*BatchMsg)(nil)
+
+func (*BatchMsg) Path() string {
+	return batch.PathExecuteBatchMsg
+}
+
+func (msg *BatchMsg) MsgList() ([]weave.Msg, error) {
+	messages := make([]weave.Msg, len(msg.Messages))
+	// make sure to cover all messages defined in protobuf
+	//TODO: Might be easier with reflection?
+	for i, m := range msg.Messages {
+		res, err := func() (weave.Msg, error) {
+			switch t := m.GetSum().(type) {
+			case *BatchMsg_Union_SendMsg:
+				return t.SendMsg, nil
+			case *BatchMsg_Union_NewTokenMsg:
+				return t.NewTokenMsg, nil
+			case *BatchMsg_Union_SetNameMsg:
+				return t.SetNameMsg, nil
+			case *BatchMsg_Union_CreateEscrowMsg:
+				return t.CreateEscrowMsg, nil
+			case *BatchMsg_Union_ReleaseEscrowMsg:
+				return t.ReleaseEscrowMsg, nil
+			case *BatchMsg_Union_ReturnEscrowMsg:
+				return t.ReturnEscrowMsg, nil
+			case *BatchMsg_Union_UpdateEscrowMsg:
+				return t.UpdateEscrowMsg, nil
+			case *BatchMsg_Union_CreateContractMsg:
+				return t.CreateContractMsg, nil
+			case *BatchMsg_Union_UpdateContractMsg:
+				return t.UpdateContractMsg, nil
+			case *BatchMsg_Union_SetValidatorsMsg:
+				return t.SetValidatorsMsg, nil
+			default:
+				return nil, errors.ErrUnknownTxType(t)
```

Comment:
pp: I am afraid we will easily forget about this function when adding new message types to protobuf. Reflection may be helpful indeed. Otherwise only high level tests would cover this. But we are in the same situation with TX type already.

Reply:
@alpe I think you are right. 
@ethanfrey would you mind if I convert this and the tx switch cases to reflection instead? it is such a pain to maintain that.

Reply:
I don't know how you intend to fix this with reflection. But if you can write and test a generic function for this in less than 2 hours, go for it. Otherwise, I don't see this as priority now, and can be made an issue for future pr.

Reply:
@ethanfrey that should be "easily" fixable via reflection, because these messages comply to the same interface and then they have 1 property each which I can get. 

I'll look into it.